File: src/types/session.ts
export interface SessionDto {
  token: string;
  user_id: number;
  username: string;
  role: 'owner' | 'pharmacist';
  full_name: string;
}

export interface LoginDto {
  username: string;
  password: string;
}

export interface SetupStatus {
  needs_setup: boolean;
}

File: src-tauri/src/models/session.rs
use serde::{Deserialize, Serialize};

/// In-memory session state stored in AppState.sessions HashMap.
/// Also persisted to the sessions SQLite table for crash recovery.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StoredSession {
    pub token: String,
    pub user_id: i64,
    pub username: String,
    pub role: String,
    pub full_name: String,
    pub created_at: String,
}

/// Session DTO returned to frontend after successful login.
/// Contains the session token, user info, but NEVER the password_hash.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionDto {
    pub token: String,
    pub user_id: i64,
    pub username: String,
    pub role: String,
    pub full_name: String,
}
